spock-core/src/main/java/org/spockframework/mock/runtime/ProxyBasedMockFactory.java

```diff
@@ -89,6 +93,9 @@ private Object createDynamicProxyMock(Class<?> mockType, List<Class<?>> addition
 
     private static final TypeCache<TypeCache.SimpleKey> CACHE =
       new TypeCache.WithInlineExpunction<>(TypeCache.Sort.SOFT);
+    private static final Random RANDOM = new Random();
```

Comment:
hm Random is not thread-safe IIRC that about ThreadLocalRandom?

Reply:
> Instances of java.util.Random are threadsafe. However, the concurrent use of the same java.util.Random instance across threads may encounter contention and consequent poor performance. Consider instead using java.util.concurrent.ThreadLocalRandom in multithreaded designs.

So it's thread-safe but may have poor performance... I'll change it to `ThreadLocalRandom`.